### PR TITLE
Py2 deprecation

### PR DESCRIPTION
**Pull request by honnix**
## Description

This consolidates all PRs for python2 deprecation: https://github.com/spotify/luigi/pulls?q=is%3Apr+label%3A%22Python2+deprecation%22+is%3Aclosed

## Motivation and Context

Open a PR to make final check easier.

### CODE REVIEW

**Comment by honnix** [comment]
cc @drowoseque . Thanks for your great contribution!

**Comment by dlstadther** [comment]
Thanks @drowoseque and @honnix for managing and doing this deprecation!

**Comment by drowoseque** [comment]
@honnix or @dlstadther could you pls resolve conflicts and merge master in this branch to keep it up-to-date?

**Comment by honnix** [comment]
@drowoseque Sure. We might get this merged next week. Sorry for taking so long.

**Comment by honnix** [comment]
@dlstadther @Tarrasch Could you please approve this as well? I will get it merged right after.

File: luigi/__meta__.py [hunk 1]
```diff
@@ -7,5 +7,5 @@
 __author__ = 'The Luigi Authors'
 __contact__ = 'https://github.com/spotify/luigi'
 __license__ = 'Apache License 2.0'
-__version__ = '2.8.13'
+__version__ = '3.0.0b2'
```
Comment (drowoseque) [inline luigi/__meta__.py:10]: @honnix shouldn't we change version here prior to merge?

Comment (honnix) [reply luigi/__meta__.py:10]: Good catch. We can release 3.0.0 immediately after merging to master. So it is OK in this PR I think.

**Comment by drowoseque** [comment]
@honnix 
btw I've found that some new code handling ImportError has arrived after merging https://github.com/spotify/luigi/pull/2845
ie 

- luigi/freezing.py:7

- luigi/tools/deps.py:48

I think we should remove it before merging
don't you mind a pr from me with eliminating that?

**Comment by honnix** [comment]
@drowoseque Sure, please go ahead.

**Comment by honnix** [comment]
`luigi/freezing.py:7` is introduced for python 3.9 #2895

**Comment by drowoseque** [comment]
okay, then no need for removal. You can go ahead :)

**Pull request merged**